### ISSUE TITLE
Fix S3927 FP: Support static abstract/virtual interface methods

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6311]
### Description

FP for S3927 (ImplementSerializationMethodsCorrectly).

Using C#11 static abstract/virtual interface methods, the rule is erroneously raising an issue.

### Repro steps

```csharp
interface IMyInterface
{
    [OnSerializing]
    static virtual void OnSerializingStaticVirtual(StreamingContext context) { } // Noncompliant FP

    [OnSerializing]
    static abstract void OnSerializingStaticAbstract(StreamingContext context); // Noncompliant FP
}
```

### Expected behavior

The rule should not raise an issue.

### Actual behavior

The rule does raise an issue.

**Comment by gregory-paidis-sonarsource** [comment]
This issue will be replaced with [this one](https://github.com/SonarSource/sonar-dotnet/issues/6331), in order to properly rework how this rule gets applied to interfaces.
